### PR TITLE
NT-1736: Migrate BackingUtils.java to kotlin ext

### PR DESCRIPTION
**Pull request by leighdouglas**
# 📲 What

- Created `BackingExt.kt` and corresponding test suite
- Refactored usages of `BackingUtils.java` methods to use new extension methods

Deletion of BackingUtils.java and class will be in subsequent PR

# 🤔 Why

Ongoing effort to migrate android to kotlin.

# 📋 QA

-

# Story 📖

[NT-1736: Migrate BackingUtils.java to kotlin ext](https://kickstarter.atlassian.net/browse/NT-1736)

### CODE REVIEW

File: app/src/main/java/com/kickstarter/libs/utils/extensions/BackingExt.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    fun Backing.isErrored(): Boolean  = this.status() == Backing.STATUS_ERRORED
+
+    fun Backing.isShippable(): Boolean {
+        val reward = this.reward() ?: return false
+        return isShippable(reward)
+    }
+
+    fun Backing.backedReward(project: Project): Reward? {
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/libs/utils/extensions/BackingExt.kt:22]: Cn we add a test for this method?

**Review by Arkariang — APPROVED** [review]
Great work here 👏  @leighdouglas . Just a small request to add tests for the method `backedRewrad`

**Comment by Arkariang** [comment]
also, will you delete `BackingUtils.java` and `BackingUtilsTest.java`  classes in another PR?

File: app/src/main/java/com/kickstarter/libs/utils/extensions/BackingExt.kt (hunk 1 above)
Comment (leighdouglas) [reply app/src/main/java/com/kickstarter/libs/utils/extensions/BackingExt.kt:22]: Whoops! You're correct. Thank you for reminding me!

**Comment by leighdouglas** [comment]
@Arkariang Yes, I like to split it up into two separate PRs

**Pull request merged**